coderd/provisionerdserver/templatescheduleoptions.go

```diff
@@ -0,0 +1,61 @@
+package provisionerdserver
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/coder/coder/coderd/database"
+)
+
+type TemplateScheduleOptions struct {
+	UserSchedulingEnabled bool          `json:"user_scheduling_enabled"`
+	DefaultTTL            time.Duration `json:"default_ttl"`
+	// If MaxTTL is set, the workspace must be stopped before this time or it
+	// will be stopped automatically.
+	//
+	// If set, users cannot disable automatic workspace shutdown.
+	MaxTTL time.Duration `json:"max_ttl"`
+}
+
+// TemplateScheduleStore provides an interface for retrieving template
+// scheduling options set by the template/site admin.
+type TemplateScheduleStore interface {
```

Comment:
~~I'm wondering why is `TemplateScheduleStore` an interface. I see only one implementation here: `agplTemplateScheduleStore`. Is it for testing purposes?~~

Ok, I see now, there is one for enterprise too.

WDYT about renaming stores to be more descriptive instead of `agpl...` and `enterprise...`. If we decide to introduce another pricing plan, then we will need to rename them accordingly :) I would move these implementations to a seperate package, maybe `schedule` or `scheduling`?

Reply:
This is done, I've moved `coderd/autobuild/schedule` to `coderd/schedule` and added the code from this file.

Reply:
This is not entirely what I meant. Currently, the `agpl...` and `enterprise...` logic are scattered. The idea is place them in the same package, but rename accordingly. These are just different strategies, so it's perfectly fine to keep them together.

For instance, `agplTemplateScheduleStore` -> `userControlledTemplateScheduleStore`, `enterpriseTemplateScheduleStore` -> `maxTTLTemplateScheduleStore`.

Reply:
We cannot place them in the same package as AGPL code can't be mixed with enterprise code, and there needs to be an implementation outside of AGPL otherwise AGPL-compiled binaries wouldn't be able to compile properly.

Reply:
I can change the names of them but I feel the names you propose are too specific especially if we add other features to them in the future.

Reply:
I see, that's unfortunate. Let's leave as is.

Reply:
This is the same pattern we use for other enterprise features that hide behind interfaces btw